projects/interfaces/filteringPipe.go

```diff
@@ -0,0 +1,30 @@
+package interfaces
+
+import (
+	"io"
+)
+
+type FilteringPipe struct {
+	PipeWriter io.Writer
+}
+
+func NewFilteringPipe(w io.Writer) *FilteringPipe {
+	return &FilteringPipe{PipeWriter: w}
+
+}
+
+func (f *FilteringPipe) Write(b []byte) (int, error) {
+	result := filterNumber(b)
+	return f.PipeWriter.Write(result)
+}
+
+var filterNumber = func (input []byte) ([]byte) {
+	var result []byte
+	for _, v:= range input{
+		if !(v >= 48 && v<= 57) {
+			result = append(result, v)
+		}
+	}
+	return result
```

Comment:
This works but it always copies the input to a new slice.

Can you think of a way of doing this that can avoid copying the data? Maybe sometimes (i.e. for specific kinds of input), or maybe always?

Reply:
I could try doing the filtering in place - moving the unwanted bytes to the far right and return the a new slice of `input[:left]`